complex/ft_test.c

```diff
@@ -269,6 +269,110 @@ static int ft_run_latency(void)
 	return 0;
 }
 
+static int ft_bw(void)
+{
+	int ret, i;
+	if (listen_sock < 0) {
+		for (i = 0; i < ft.xfer_iter; i++) {
+                        ret = ft_send_msg();
+			if (ret)
+				return ret;
+		}
+                while (ft_tx.credits < ft_tx.max_credits) {
+                        ret = ft_comp_tx();
+                        if (ret)
+                                return ret;
+                }
+                ret = ft_recv_msg();
+                if (ret)
+			return ret;
+	} else {
+		int count=0;
+                do {
+			ret = ft_post_recv_bufs();
+			if (ret)
+				return ret;
+			ret = ft_comp_rx();
+			if (ret)
+				return ret;
+			count+=ft_rx.credits;
+                } while (count < ft.xfer_iter);
+                ret = ft_send_msg();
+                if (ret)
+			return ret;
+                while (ft_tx.credits < ft_tx.max_credits) {
+			ret = ft_comp_tx();
+			if (ret)
+				return ret;
+                }
+	}
+	return 0;
+}
+
+static int ft_bw_dgram(void)
+{
+	int ret, i;
+
+	if (listen_sock < 0) {
+		for (i = 0; i < ft.xfer_iter; i++) {
+			ret = ft_send_dgram();
+			if (ret)
+				return ret;
+		}
+		ret = ft_send_dgram_done();
+		if(ret)
+			return ret;
+		int credits = 0;
+		while(credits)
```

Comment:
I intended this loop to refill the receive credits for the ack, but it’s obviously not needed.
Sendrecv_dgram (replaces the send_dgram_done) refills the receive credits when 1/2 are used.


> On Apr 8, 2015, at 2:36 PM, Sean Hefty <notifications@github.com> wrote:
> 
> In complex/ft_test.c <https://github.com/ofiwg/fabtests/pull/221#discussion_r28016221>:
> 
> > +
> > +static int ft_bw_dgram(void)
> > +{
> > +	int ret, i;
> > +
> > +	if (listen_sock < 0) {
> > +		for (i = 0; i < ft.xfer_iter; i++) {
> > +			ret = ft_send_dgram();
> > +			if (ret)
> > +				return ret;
> > +		}
> > +		ret = ft_send_dgram_done();
> > +		if(ret)
> > +			return ret;
> > +		int credits = 0;
> > +		while(credits)
> This while loop is never executed.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/ofiwg/fabtests/pull/221/files#r28016221>.
>